src/main/java/edu/cornell/kfs/module/ar/batch/service/impl/CuVerifyBillingFrequencyServiceImpl.java

```diff
@@ -69,31 +72,45 @@ private Date calculateLastBilledDateBasedOnAwardInvoiceOption(ContractsAndGrants
+     */
     @Override
-    public boolean validateBillingFrequency(ContractsAndGrantsBillingAward award, ContractsAndGrantsBillingAwardAccount awardAccount) {
-        LOG.info("validateBillingFrequency: CU Customization invoked with Award, Award Account. Sending awardAccount = "
-                + awardAccount.getAccountNumber() + " with currentLastBilledDate = "
-                + awardAccount.getCurrentLastBilledDate() + " and Award containing creationProcessTypeCode = "
+    public boolean validateBillingFrequency(ContractsAndGrantsBillingAward award, ContractsAndGrantsBillingAwardAccount awardAccount, boolean checkBillingPeriodEnd) {
+        LOG.info("validateBillingFrequency: CU Customization invoked with Award, Award Account and boolean. Sending awardAccount = "
+                + awardAccount.getAccountNumber() + " with currentLastBilledDate = " 
+                + awardAccount.getCurrentLastBilledDate() + ", checkBillingPeriodEnd = " + checkBillingPeriodEnd
+                + " and Award containing creationProcessTypeCode = "
                 + ArConstants.ContractsAndGrantsInvoiceDocumentCreationProcessType.getName(award.getCgInvoiceDocumentCreationProcessTypeCode()) + " to private method.");
-        return validateBillingFrequency(award, awardAccount.getCurrentLastBilledDate());
+        return validateBillingFrequency(award, awardAccount.getCurrentLastBilledDate(), checkBillingPeriodEnd);
     }
 
-    private boolean validateBillingFrequency(ContractsAndGrantsBillingAward award, Date lastBilledDate) {
+    /*
+     * CUMod: KFSPTS-15655 additional logging
+     * CUMod: KFSPTS-14970 
+     */
+    //CUMod: Super class private method with additional logging and 
+    private boolean validateBillingFrequency(ContractsAndGrantsBillingAward award, Date lastBilledDate, boolean checkBillingPeriodEnd) {
         LOG.info("validateBillingFrequency(private): For Award/Proposal# = " + award.getProposalNumber()
-                + " with calculatedLastBilledDate = " + lastBilledDate
+                + " with calculatedLastBilledDate = " + lastBilledDate + ", checkBillingPeriodEnd = " + checkBillingPeriodEnd
                 + " creationProcessTypeCode = " + ArConstants.ContractsAndGrantsInvoiceDocumentCreationProcessType.getName(award.getCgInvoiceDocumentCreationProcessTypeCode()));
         final Date today = getDateTimeService().getCurrentSqlDate();
         AccountingPeriod currPeriod = getAccountingPeriodService().getByDate(today);
 
-        BillingPeriod billingPeriod = getStartDateAndEndDateOfPreviousBillingPeriod(award, lastBilledDate, currPeriod, award.getCgInvoiceDocumentCreationProcessTypeCode());
+        //basecode: BillingPeriod billingPeriod = getStartDateAndEndDateOfPreviousBillingPeriod(award, currPeriod);
+        BillingPeriod billingPeriod = getStartDateAndEndDateOfPreviousBillingPeriod(award, lastBilledDate, currPeriod, award.getCgInvoiceDocumentCreationProcessTypeCode()); //CUmod
```

Comment:
if the cumod comment is needed, I would move that to it's own line above

Reply:
Moved.